massa-execution-exports/src/controller_traits.rs

```diff
@@ -19,12 +23,12 @@ pub trait ExecutionController: Send + Sync {
     /// Updates blockclique status by signaling newly finalized blocks and the latest blockclique.
     ///
     /// # Arguments
-    /// * `finalized_blocks`: newly finalized blocks
-    /// * `blockclique`: new blockclique
+    /// * `finalized_blocks`: newly finalized blocks. Each Storage owns refs to the block and its ops/endorsements/parents.
+    /// * `blockclique`: new blockclique. Each Storage owns refs to the block and its ops/endorsements/parents
     fn update_blockclique_status(
         &self,
-        finalized_blocks: HashMap<Slot, BlockId>,
-        blockclique: HashMap<Slot, BlockId>,
+        finalized_blocks: HashMap<Slot, (BlockId, Storage)>,
+        blockclique: HashMap<Slot, (BlockId, Storage)>,
```

Comment:
Maybe use a global storage for all that stuff would be better. I feel like we need to write-lock the owned variable for each id. In that case, with a global storage we could lock only once.

Reply:
Yes but I'm afraid that would imply losing the per-block storage that can be moved around internally by the execution worker.